repos/system_upgrade/el7toel8/actors/checkpcidrivers/tests/unit_test_checkpcidrivers.py

```diff
@@ -0,0 +1,367 @@
+            available_rhel8=1,
+            supported_rhel8=0,
+            available_rhel9=0,
+            supported_rhel9=0,
+            comment="nan",
+        ),
+        RestrictedPCIDevice(
+            pci_id="0x1000:0x0065",
+            driver_name="mpt2sas",
+            device_name="SAS2116_2",
+            available_rhel7=1,
+            supported_rhel7=1,
+            available_rhel8=1,
+            supported_rhel8=1,
+            available_rhel9=0,
+            supported_rhel9=0,
+            comment="nan",
+        ),
+    ],
+)
+
+
+@pytest.mark.parametrize(
+    (
+        "driver_name",
+        "pci_id",
+        "inhibit_upgrade",
+        "num_msgs",
+    ),
+    [
+        (
+            "good driver name",
+            "good pci id",
+            None,
+            0,
+        ),
+        (
+            "3w-9xxx",
+            "good pci id",
+            True,
+            1,
+        ),
+        (
+            "3w-sas",
+            "good pci id",
+            None,
+            1,
+        ),
+        (
+            "acard-ahci",
+            "good pci id",
+            None,
+            0,
+        ),
+        (
+            "good driver name",
+            "0x1000:0x0060",
+            True,
+            1,
+        ),
+        (
+            "good driver name",
+            "0x1000:0x0064",
+            None,
+            1,
+        ),
+        (
+            "good driver name",
+            "0x1000:0x0065",
+            None,
+            0,
+        ),
+    ],
+)
+def test_basic_checkpcidrivers(
+    monkeypatch,
+    driver_name,
+    pci_id,
+    inhibit_upgrade,
+    num_msgs,
+):
+    """Check the main actor function."""
+    pci_devices.devices[0].driver = driver_name
+    pci_devices.devices[0].pci_id = pci_id
+    monkeypatch.setattr(
+        api,
+        "current_actor",
+        CurrentActorMocked(msgs=[pci_devices, restricted_pci_devices]),
+    )
+    monkeypatch.setattr(
+        reporting,
+        "create_report",
+        create_report_mocked(),
+    )
+    monkeypatch.setattr(
+        CurrentActorMocked,
+        "produce",
+        produce_mocked(),
+    )
+    checkpcidrivers_main()
+    if inhibit_upgrade:
+        assert len(CurrentActorMocked.produce.model_instances) == 1
+        assert CurrentActorMocked.produce.model_instances[0].report[
+            "flags"
+        ] == ["inhibitor"]
+    if num_msgs and not inhibit_upgrade:
+        assert (
+            "flags" not in CurrentActorMocked.produce.model_instances[0].report
+        )
+    assert len(CurrentActorMocked.produce.model_instances) == num_msgs
+
+
+RestrictedPCIDevicePrefilled = partial(
+    RestrictedPCIDevice,
+    pci_id="",
+    driver_name="",
+    device_name="",
+    comment="",
+    available_rhel7=1,
+    available_rhel9=0,
+    supported_rhel7=1,
+    supported_rhel9=0,
+)
+
+
+def get_test_case_for_testing_report(
+    devices, available_rhel8, supported_rhel8
+):
+    return tuple(d for d in devices), {
```

Comment:
This tuple conversion isn't necessary, you have it already being a tuple but it's not a blocker

Reply:
Changed just to `devices`:) thx for catching this